lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp

```diff
@@ -0,0 +1,195 @@
+            SYS->wSecond = TM->tm_hour
+
+    #define SYSTIME_TO_TM(SYS, TM)              \
+            TM->tm_year = SYS->wYear - 1900; \
+            TM->tm_mon = SYS->wMonth - 1;  \
+            TM->tm_wday = SYS->wDayOfWeek;   \
+            TM->tm_mday = SYS->wDay;         \
+            TM->tm_hour = SYS->wHour;        \
+            TM->tm_min = SYS->wMinute;       \
+            TM->tm_sec = SYS->wSecond
+
+    const bool SysLocalToUtc(SYSTEMTIME *local, SYSTEMTIME *utc)
+    {
+        struct tm local_tm;
+        SYSTIME_TO_TM(local, (&local_tm));
+
+        // tm doesn't have milliseconds
+        // uint32_t msec = utc->wMilliseconds;
+        // simply keep it as is
+
+        time_t utime = mktime(&local_tm);
+        
+        struct tm *utc_tm;
+        utc_tm = gmtime (&utime);
+
+        TM_TO_SYSTIME(utc, utc_tm);
+
+        // put milliseconds back
+        utc->wMilliseconds = local->wMilliseconds;
+
+        return true;
+    }
+
+    const bool SysUtcToLocal(SYSTEMTIME *utc, SYSTEMTIME *local, int64_t &bias, int &offset, bool &isDaylightSavings)
+    {
+        struct tm utc_tm;
+        SYSTIME_TO_TM(utc, (&utc_tm));
+
+        // tm doesn't have milliseconds
+        // uint32_t msec = utc->wMilliseconds;
+        // simply keep it as is
+
+        time_t ltime = mktime(&utc_tm);
+        
+        struct tm local_tm;
+        
+        tzset();
+        localtime_r(&ltime, &local_tm);
+        TM_TO_SYSTIME(local, (&local_tm));
+
+        // put milliseconds back
+        local->wMilliseconds = utc->wMilliseconds;
+
+        isDaylightSavings = local_tm.tm_isdst;
+
+        offset = local_tm.tm_gmtoff / 60;
+
+        struct tm *gtime = gmtime(&ltime);
+        time_t mtime = mktime(gtime);
+
+        bias = (int64_t)(mtime - ltime) / 60;
+
+        return true;
+    }
+
+    // DaylightTimeHelper ******
+    static void YmdToSystemTime(YMD* ymd, SYSTEMTIME *sys) noexcept
+    {
+        sys->wYear = (WORD)ymd->year;
+        sys->wMonth = (WORD)(ymd->mon + 1);
+        sys->wDay =(WORD)(ymd->mday + 1);
+        int t = ymd->time;
+        sys->wMilliseconds = (WORD)(t % 1000);
+        t /= 1000;
+        sys->wSecond = (WORD)(t % 60);
+        t /= 60;
+        sys->wMinute = (WORD)(t % 60);
+        t /= 60;
+        sys->wHour = (WORD)t;
+    }
+
+    
+    const WCHAR *DaylightTimeHelper::GetStandardName(size_t *nameLength) {
+        UpdateTimeZoneInfo();
+        *nameLength = zoneNameLength;
+        return zoneName;
+    }
+
+    const WCHAR *DaylightTimeHelper::GetDaylightName(size_t *nameLength) {
+        // We have the abbreviated standard or daylight name
+        // based on date and time zone we are in.
+        return GetStandardName(nameLength);
+    }
+
+    double DaylightTimeHelper::UtcToLocal(double utcTime, int &bias, int &offset, bool &isDaylightSavings)
+    {
+        SYSTEMTIME utcSystem, localSystem;
+        YMD ymd;
+
+        Js::DateUtilities::GetYmdFromTv(utcTime, &ymd);
```

Comment:
Should be fine for now but it seems like this can be optimized- we probably don't need to convert to YMD and then to SYSTEMTIME etc- seems like on linux, we should have a double => tm and tm => double conversion function. Then we should be able to simplify this code significantly- since it looks like YMD and tm have a lot of overlap. Alternatively, we could probably do double => time_t and do the same utc/local conversions. But for now, the current implementation is fine.